### PR TITLE
Add testing of grpc-node for Linux and MacOS

### PR DESCRIPTION
**Pull request by matt-kwong**
Testing for now

### CODE REVIEW

File: tools/run_tests/helper_scripts/run_grpc-node.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+# cd to gRPC root directory
+cd $(dirname $0)/../../..
+
+CURRENT_COMMIT=$(git rev-parse --verify HEAD)
+
+git clone --recursive https://github.com/grpc/grpc-node ./../grpc-node
+cd ./../grpc-node
+
+test-grpc-submodule.sh $CURRENT_COMMIT
```
Comment (nicolasnoble) [inline tools/run_tests/helper_scripts/run_grpc-node.sh]: I think you want to add ./ here - relying on '.' to be part of the PATH isn't really reliable.

Comment (matt-kwong) [reply tools/run_tests/helper_scripts/run_grpc-node.sh]: Done

**Comment by matt-kwong** [comment]
@murgatroid99 can you look at the grpc-node failures in MacOS and Linux? It looks like a build failure

**Comment by matt-kwong** [comment]
Ping, the error is this for Linux:
```
01:50:27.830 Error: /var/local/git/grpc-node/packages/grpc-native-core/src/node/extension_binary/node-v57-linux-x64/grpc_node.node: undefined symbol: grpc_stream_compression_gzip_vtable
01:50:27.830     at Object.Module._extensions..node (module.js:653:18)
01:50:27.830     at Module.load (module.js:545:32)
01:50:27.831     at tryModuleLoad (module.js:508:12)
01:50:27.831     at Function.Module._load (module.js:500:3)
01:50:27.831     at Module.require (module.js:568:17)
01:50:27.831     at require (internal/module.js:11:18)
01:50:27.831     at Object.<anonymous> (/var/local/git/grpc-node/packages/grpc-native-core/src/grpc_extension.js:30:15)
01:50:27.831     at Module._compile (module.js:624:30)
01:50:27.831     at Object.Module._extensions..js (module.js:635:10)
01:50:27.831     at Module.load (module.js:545:32)
01:50:27.831     at tryModuleLoad (module.js:508:12)
01:50:27.831     at Function.Module._load (module.js:500:3)
01:50:27.831     at Module.require (module.js:568:17)
01:50:27.831     at require (internal/module.js:11:18)
01:50:27.831     at Object.<anonymous> (/var/local/git/grpc-node/packages/grpc-native-core/src/client.js:38:12)
01:50:27.831     at Module._compile (module.js:624:30)
01:50:27.831     at Object.Module._extensions..js (module.js:635:10)
01:50:27.831     at Module.load (module.js:545:32)
01:50:27.831     at tryModuleLoad (module.js:508:12)
01:50:27.831     at Function.Module._load (module.js:500:3)
01:50:27.831     at Module.require (module.js:568:17)
01:50:27.831     at require (internal/module.js:11:18)
01:50:27.831     at Object.<anonymous> (/var/local/git/grpc-node/packages/grpc-native-core/index.js:30:14)
01:50:27.831     at Module._compile (module.js:624:30)
01:50:27.831     at Object.Module._extensions..js (module.js:635:10)
01:50:27.831     at Module.load (module.js:545:32)
01:50:27.831     at tryModuleLoad (module.js:508:12)
01:50:27.831     at Function.Module._load (module.js:500:3)
01:50:27.831     at Module.require (module.js:568:17)
01:50:27.831     at require (internal/module.js:11:18)
01:50:27.831     at Object.<anonymous> (/var/local/git/grpc-node/test/api/credentials_test.js:25:12)
01:50:27.831     at Module._compile (module.js:624:30)
01:50:27.831     at Object.Module._extensions..js (module.js:635:10)
01:50:27.831     at Module.load (module.js:545:32)
01:50:27.831     at tryModuleLoad (module.js:508:12)
01:50:27.831     at Function.Module._load (module.js:500:3)
01:50:27.831     at Module.require (module.js:568:17)
01:50:27.831     at require (internal/module.js:11:18)
01:50:27.831     at /var/local/git/grpc-node/node_modules/mocha/lib/mocha.js:231:27
01:50:27.831     at Array.forEach (<anonymous>)
01:50:27.831     at Mocha.loadFiles (/var/local/git/grpc-node/node_modules/mocha/lib/mocha.js:228:14)
01:50:27.831     at Mocha.run (/var/local/git/grpc-node/node_modules/mocha/lib/mocha.js:514:10)
01:50:27.831     at Object.<anonymous> (/var/local/git/grpc-node/node_modules/mocha/bin/_mocha:480:18)
01:50:27.831     at Module._compile (module.js:624:30)
01:50:27.831     at Object.Module._extensions..js (module.js:635:10)
01:50:27.831     at Module.load (module.js:545:32)
01:50:27.831     at tryModuleLoad (module.js:508:12)
01:50:27.831     at Function.Module._load (module.js:500:3)
01:50:27.831     at Function.Module.runMain (module.js:665:10)
01:50:27.831     at startup (bootstrap_node.js:201:16)
01:50:27.831     at bootstrap_node.js:626:3
01:50:27.831 
01:50:27.831 Details:
01:50:27.831     code: 1
01:50:27.831     killed: false
01:50:27.831     stdout: 
01:50:27.831     stderr: module.js:653
01:50:27.831   return process.dlopen(module, path._makeLong(filename));
```

**Comment by murgatroid99** [comment]
It looks like the template code failed to run in the grpc-node repo. And the symbol that was not found appears to be in a file that was added since the commit that grpc-node was pinned to, so the template script would have to succeed to add that file to the list in `binding.gyp`.

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__2M_2_0.counters.new: 1
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__2M_2_0.counters.old: 1


[microbenchmarks] Performance differences noted:
Benchmark                                                              cpu_time
---------------------------------------------------------------------  ----------
BM_StreamingPingPong<InProcessCHTTP2, NoOpMutator, NoOpMutator>/16M/2  +4%
```

**Pull request merged**